cdap-client/src/main/java/co/cask/cdap/client/ApplicationClient.java

```diff
@@ -179,6 +179,29 @@ public ApplicationDetail get(Id.Application appId)
   }
 
   /**
+   * Get details about the specified application.
+   *
+   * @param appId the id of the application to get
+   * @return details about the specified application
+   * @throws ApplicationNotFoundException if the application with the given ID was not found
+   * @throws IOException if a network error occurred
+   * @throws UnauthenticatedException if the request is not authorized successfully in the gateway server
+   */
+  public ApplicationDetail get(ApplicationId appId)
```

Comment:
shouldn't have two ``get`` and two ``delete`` methods like this. I am not sure about backward compatibility - unless we want to maintain it, the old ``get`` and ``delete`` methods should be removed in favor of the new ones. If we want backward compatibility, they should be deprecated in favor of these methods.